src/main/java/com/cleanroommc/groovyscript/compat/mods/extrautils2/GridPowerPassiveGenerator.java

```diff
@@ -0,0 +1,120 @@
+            GeneratorTypeAccessor accessor = (GeneratorTypeAccessor) value;
+            float[] capsInput = scalingMap.get(accessor.getKey());
+            if (capsInput == null) return;
+            float[][] caps = new float[capsInput.length / 2][2];
+
+            for (int i = 0; i < capsInput.length; i += 2) {
+                caps[i / 2][0] = capsInput[i];
+                caps[i / 2][1] = capsInput[i + 1];
+            }
+            accessor.setCaps(IWorldPowerMultiplier.createCapsTree(caps));
+        }
+    }
+
+    @MethodDescription(type = MethodDescription.Type.VALUE)
+    public void setPowerMultiplier(BlockPassiveGenerator.GeneratorType generator, IWorldPowerMultiplier worldPowerMultiplier) {
+        addScripted(Pair.of(generator, worldPowerMultiplier));
+        ((GeneratorTypeAccessor) generator).setPowerMultiplier(worldPowerMultiplier);
+    }
+
+    @MethodDescription(example = @Example(value = "resource('generators:wind'), IWorldPowerMultiplier.CONSTANT", imports = "com.rwtema.extrautils2.power.IWorldPowerMultiplier"), type = MethodDescription.Type.VALUE)
+    public void setPowerMultiplier(ResourceLocation generator, IWorldPowerMultiplier worldPowerMultiplier) {
+        Arrays.stream(BlockPassiveGenerator.GeneratorType.values())
+                .filter(x -> ((GeneratorTypeAccessor) x).getKey().equals(generator))
+                .findFirst()
+                .ifPresent(x -> setPowerMultiplier(x, worldPowerMultiplier));
+    }
+
+    @MethodDescription(type = MethodDescription.Type.VALUE)
+    public void setPowerMultiplier(String generator, IWorldPowerMultiplier worldPowerMultiplier) {
+        setPowerMultiplier(new ResourceLocation(generator), worldPowerMultiplier);
+    }
+
+    @MethodDescription(type = MethodDescription.Type.VALUE,
+                       example = @Example(value = "resource('generators:solar'), { TilePassiveGenerator generator, World world -> 100f }", imports = "com.rwtema.extrautils2.tile.TilePassiveGenerator"))
+    public void setPowerLevel(ResourceLocation generator, Closure<Float> powerLevel) {
+        if (powerLevel == null) {
+            GroovyLog.msg("Extra Utilities 2 Grid Power Passive Generator powerLevel closure must be defined")
+                    .error()
+                    .post();
+            return;
+        }
+        if (Arrays.equals(powerLevel.getParameterTypes(), new Class[]{TilePassiveGenerator.class, World.class})) {
+            this.powerLevelMap.put(generator, powerLevel);
+            return;
+        }
+        GroovyLog.msg("Extra Utilities 2 Grid Power Passive Generator powerLevel closure requires a closure with exactly two parameters:")
+                .add("com.rwtema.extrautils2.tile.TilePassiveGenerator generator, net.minecraft.world.World world in that order.")
+                .add("but had {}, {} instead", (Object[]) powerLevel.getParameterTypes())
+                .error()
+                .post();
+    }
```

Comment:
We are not forcing param types on closures. Only log an error in debug mode. Otherwise dont do anything.